drools-game-engine-capture-flag-kjar/src/main/java/org/drools/game/capture/flag/cmds/PickFlagCommand.java

```diff
@@ -37,7 +43,22 @@ public PickFlagCommand( Player player, Flag flag ) {
     public Void execute( Context ctx ) {
         KieSession session = ( KieSession ) ctx.getData().get( "session" );
         GameMessageService messageService = ( GameMessageService ) ctx.getData().get( "messageService" );
+        
         FactHandle playerFH = session.getFactHandle( getPlayer());
+
+//        FactHandle playerFH = session.getFactHandles(new ObjectFilter() {
```

Comment:
Just remove the code if it is not needed anymore. I notice that you are using equality now, so now equals() and hashcode() methods needs to be propertly implented in all the game Facts.